Parser/action_helpers.c

```diff
@@ -1691,3 +1691,18 @@ _PyPegen_concatenate_strings(Parser *p, asdl_expr_seq *strings,
     assert(current_pos == n_elements);
     return _PyAST_JoinedStr(values, lineno, col_offset, end_lineno, end_col_offset, p->arena);
 }
+
+stmt_ty
+_PyPegen_checked_future_import(Parser *p, identifier module, asdl_alias_seq * names, int level,
+                  			   int lineno, int col_offset, int end_lineno, int end_col_offset,
+                      		   PyArena *arena) {
+    if (PyUnicode_CompareWithASCIIString(module, "__future__") == 0) {
```

Comment:
I'm just wondering but what happens if you have a fake `__future__` module with a fake `barry_as_FLUFL`. Would it actually enable the flag? (I don't think we are forbidden to create our own `__future__` module right? AFAIK, it follows the same rules as a regular module).

Reply:
Yes, that would work. The effect is syntactic; it does also import the module at runtime but the future import is activated by the parser, not by the specific contents of `__future__.py`.

Reply:
This is the expectation. The fact that there is a backing module after that import is just an implementation detail.

Reply:
Does it activate it here? See #118216.

> ```python
> >>> from .__future__ import barry_as_FLUFL
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>     from .__future__ import barry_as_FLUFL
> ImportError: attempted relative import with no known parent package
> >>> 1 <> 2
> True
> ```